### PR TITLE
Read GYRO and ACC in one transaction and only use acc when needed.

### PR DESCRIPTION
**Pull request by hydra**
Experimental code that allows the ACC to be enabled and read as soon as

the MPU signals that new data is ready but stored until actually needed.

This is proof-of-concept code and not really sure it's particularly
worthwhile at this time.  Just wanting some feedback as this has been
discussed in the past.

### CODE REVIEW

**Comment by hydra** [comment]
this flies, but after a while it builds up a forwards and left drift which gets worse over time.  likely due to the times that the acc data being read is different to the values uses by the imu.

needs more work, but you guys get the idea right?

File: src/main/flight/imu.c [hunk 1]
```diff
@@ -210,9 +209,21 @@ static float invSqrt(float x)
     return 1.0f / sqrtf(x);
 }
 
+static uint32_t imuGainResetTime = 0;
+
+void imuResetGainTime(uint32_t millis)
+{
+    imuGainResetTime = millis;
```
Comment (ledvinap) [inline src/main/flight/imu.c]: local parameter is shadowing function

File: src/main/drivers/accgyro_mpu.c [hunk 2]
```diff
@@ -305,34 +331,54 @@ static bool mpuWriteRegisterI2C(uint8_t reg, uint8_t data)
     return ack;
 }
 
+static void convertAccBufferToAccData(int16_t *accData, uint8_t *buffer)
```
Comment (hydra) [inline src/main/drivers/accgyro_mpu.c]: yup, not a bad idea that.

**Comment by ledvinap** [comment]
Also - there should be good reason to skip attitude update.

IMU will take some time to converge. If switching to ANGLE/HORIZON as failsafe, there will be significant acceleration. Either ACC will be out of range (and there will be zero position update), or ADD reading will be low quality. PID regulators will act on bad attitude estimate and probably make situation even worse.

The situation will be similar even when attitude is updated in ACRO mode and only ACC gain is increased.

What you want is to set attitude confidence to (almost) zero on GYRO saturation, use attitude confidence when calculating ACC gain. And possibly modify PID gains when attitude confidence is low (the system is closed, it may get unstable during this phase)

**Comment by borisbstyle** [comment]
Nice. Will play with this soon!

**Comment by hydra** [comment]
@borisbstyle did you have a look at this?

**Comment by hydra** [comment]
I will close this to de-clutter the issue tracker, it was experimental and requires additional work and is out of date too.

**Pull request closed without merging**